dev/droidmate/projects/core/src/main/groovy/org/droidmate/android_sdk/AdbWrapper.groovy

```diff
@@ -573,30 +581,31 @@ public class AdbWrapper implements IAdbWrapper
     {
       String commandDescription = String
         .format(
-        "Executing adb to start UiAutomatorDaemon.init() method on Android Device with " +
+        "Executing adb to start UiAutomatorDaemon service on Android Device with " +
           "s/n %s.",
         deviceSerialNumber)
 
-      String uiaDaemonCmdLine = String.format("-c %s -e %s %s -e %s %s -e %s %s",
-        UiautomatorDaemonConstants.uiaDaemon_initMethodName,
+      String uiaDaemonCmdLine = String.format("-e %s %s -e %s %s -e %s %s",
         UiautomatorDaemonConstants.uiaDaemonParam_waitForGuiToStabilize, cfg.uiautomatorDaemonWaitForGuiToStabilize,
         UiautomatorDaemonConstants.uiaDaemonParam_waitForWindowUpdateTimeout, cfg.uiautomatorDaemonWaitForWindowUpdateTimeout,
         UiautomatorDaemonConstants.uiaDaemonParam_tcpPort, port)
 
       this.sysCmdExecutor.executeWithoutTimeout(commandDescription, cfg.adbCommand,
         "-s", deviceSerialNumber,
-        "shell uiautomator runtest",
-        cfg.uiautomatorDaemonJar.fileName.toString(),
-        uiaDaemonCmdLine)
+        "shell am instrument",
+        "--user 0",
+        uiaDaemonCmdLine,
+        "-w",
+        UiautomatorDaemonConstants.uiaDaemon_testPackageName + "/" + UiautomatorDaemonConstants.uiaDaemon_testRunner)
 
     } catch (SysCmdExecutorException e)
     {
-      throw new AdbWrapperException("Executing 'adb shell uiautomator runtest ...' failed. Oh my.", e)
+      throw new AdbWrapperException("Executing 'adb shell instrument --user 0 -w org.droidmate.uiautomator2daemon ...' failed. Oh my.", e)
     }
   }
 
   @Override
-  void removeFile(String deviceSerialNumber, String fileName) throws AdbWrapperException
+  void removeFile(String deviceSerialNumber, String fileName, String runAsPackage) throws AdbWrapperException
```

Comment:
"runAsPackage" is confusing. It looks to be a Boolean "should I run as package, or not?" but instead it denotes a package being an argument to "run-as" command. Consider renaming to "shellPackage" + add in method doc link to official documentation and/or SO Q&A what is this "run-as" (I have no idea why this is necessary).